src/pantry.js

```diff
@@ -0,0 +1,69 @@
+class Pantry {
+  constructor(ingredients) {
+    this.ingredients = this.clean(ingredients);
+  }
+
+  clean(data) {
+    let cleanedData = data.reduce((list, item) => {
+      let exists = list.find(i => i.ingredient === item.ingredient);
+      let index = list.indexOf(exists);
+      exists?
+        list[index].amount += item.amount:
+        list.push(item);
+      return list;
+    }, [])
+    return cleanedData;
+  }
```

Comment:
Nice job putting this together! Hopefully this makes the other fetch related functions go a little smoother!